.circleci/config.yml

```diff
@@ -865,31 +963,45 @@ jobs:
       - *attach_workspace
       - *add_ssh_keys
       - *prepare_environment
-      - run:
-          name: Check User Permissions to Force Upload Pack
-          command: |
-            CONTENT_LEADERS=$(curl -sS https://api.github.com/orgs/demisto/teams/content-leaders/members -H "Authorization: token ${GITHUB_TOKEN}")
-            LEADER_NAMES=$(echo $CONTENT_LEADERS | jq -r ".[].login")
-            LEADER_NAMES=$(echo "${LEADER_NAMES}" "content-bot" )
-            if [[ -z "${CIRCLE_USERNAME}" ]] || [[ -z "`echo $LEADER_NAMES | grep -w "${CIRCLE_USERNAME}"`" ]]; then
-                echo -e "User '${CIRCLE_USERNAME}' is not allowed to trigger this build, only one of:\n${LEADER_NAMES}"
-                exit 1
-            else
-                echo "User '${CIRCLE_USERNAME}' is allowed to force pack uploads."
-            fi
+      - *check_user_permissions_to_upload_packs
+      - *force_upload_packs_to_marketplace_storage
+      - *destroy_instances
+      - *store_artifacts
 
+  Upload Packs To Marketplace:
+    <<: *container_config
+    resource_class: medium
+    <<: *environment
+    steps:
+      - *attach_workspace
+      - *add_ssh_keys
+      - *prepare_environment
+      - *check_user_permissions_to_upload_packs
+      - when:
+          condition:
+            equal:
+              - master
+              - << pipeline.git.branch >>
+          steps:
+            - run:
+                name: Upload Packs To Marketplace Storage
+                command: |
+                  EXTRACT_FOLDER=$(mktemp -d)
+                  PACK_ARTIFACTS=$CIRCLE_ARTIFACTS/content_packs.zip
+                  PACKS_DEPENDENCIES=$CIRCLE_ARTIFACTS/packs_dependencies.json
+                  ID_SET=$CIRCLE_ARTIFACTS/id_set.json
+                  GCS_PATH=$(mktemp)
+                  CIRCLE_BRANCH=${CIRCLE_BRANCH:-unknown}
+                  CREATE_INSTANCES_JOB_NUMBER=$(cat create_instances_build_num.txt)
+                  echo $GCS_MARKET_KEY > $GCS_PATH
+                  python3 ./Tests/Marketplace/copy_and_upload_packs.py -a $PACK_ARTIFACTS -e $EXTRACT_FOLDER -pb 'marketplace-dist' -bb 'marketplace-ci-build' -i $ID_SET -s $GCS_PATH -n $CREATE_INSTANCES_JOB_NUMBER -c $CIRCLE_BRANCH -o false
```

Comment:
From what source are you copying? 
```
marketplace-ci-build/content/builds/master
```
?

Reply:
yes, refactored it to be the same as upload